sqlmesh/lsp/api.py

```diff
@@ -0,0 +1,70 @@
+"""
+This module maps the LSP custom API calls to the SQLMesh web api.
+
+Allowing the LSP to call the web api without having to know the details of the web api
+and thus passing through the details of the web api to the LSP, so that both the LSP
+and the web api can communicate with the same process, avoiding the need to have a
+separate process for the web api.
+"""
+
+import typing as t
+from sqlmesh.utils.pydantic import PydanticModel
+from web.server.models import Model
+
+API_FEATURE = "sqlmesh/api"
+
+
+class ApiRequest(PydanticModel):
+    """
+    Request to call the SQLMesh API.
+    This is a generic request that can be used to call any API endpoint.
+    """
+
+    requestId: str
+    url: str
+    method: t.Optional[str] = "GET"
+    params: t.Optional[t.Dict[str, t.Any]] = None
+    body: t.Optional[t.Dict[str, t.Any]] = None
+
+
+class ApiResponse(PydanticModel):
+    """
+    Response from the SQLMesh API.
+    This is a generic base class that can be used to return data from any API endpoint.
+    Specific API responses should inherit from this class and specify the data type more precisely.
+    """
+
+    data: t.Union[t.Dict[str, t.Any], t.List[t.Any]]
+
+
+class ApiResponseGetModels(ApiResponse):
+    """
+    Response from the SQLMesh API for the get_models endpoint.
+    Specifies the data type more precisely as a list of models.
+    """
+
+    data: t.List[Model]
+
+    def __init__(self, **data: t.Any) -> None:
+        # Convert datetime objects to strings before passing to parent constructor
+        if "data" in data and isinstance(data["data"], list):
+            for model in data["data"]:
+                if hasattr(model, "details") and model.details:
+                    # Convert datetime fields to None or string to avoid serialization issues
+                    for field in ["stamp", "start", "cron_prev", "cron_next"]:
```

Comment:
What are these fields? Only `start` here can actually be datetime

Reply:
https://github.com/TobikoData/sqlmesh/blob/1b073394b9c5c84f4be4173e9d50b99bd23f3cf5/web/server/models.py#L143

Reply:
It seems like these were [changed](https://github.com/TobikoData/sqlmesh/pull/854) quite some time ago. I can tell for certain that `stamp` is supposed to be a string ([docs](https://sqlmesh.readthedocs.io/en/stable/concepts/models/overview/?h=stamp#stamp)), so I wonder if the change was an oversight. The cron prev/next do [look](https://github.com/TobikoData/sqlmesh/blob/e773b59a626cc7607930f014f045c78f60add72f/sqlmesh/core/node.py#L103-L127) like datetime.